src/main/java/org/prebid/server/bidder/pubnative/PubnativeBidder.java

```diff
@@ -85,9 +92,22 @@ private ExtImpPubnative parseImpExt(ObjectNode impExt) {
         }
     }
 
-    private static BidRequest modifyRequest(BidRequest bidRequest, Imp imp) {
-        Imp outgoingImp = imp;
-        final Banner banner = imp.getBanner();
+    private BidRequest modifyRequest(BidRequest bidRequest, Imp imp) {
+        final BigDecimal resolvedBidFloor = resolveBidFloor(bidRequest, imp);
+        final String resolvedBidFloorCur = resolvedBidFloor != null ? CUR_USD : null;
+        final Imp outgoingImp = imp.toBuilder()
```

Comment:
Creating new instance if `Imp` should be conditional if `bidfloor` is changed since it is quite expensive opertaion.
PS. the same for re-creating bid request below.

Reply:
We cannot make re-creating bid request conditional as we need to leave only one imp in corresponding request.